control-plane-operator/main.go

```diff
@@ -112,13 +111,14 @@ func NewStartCommand() *cobra.Command {
 			Port:                   9443,
 			LeaderElection:         enableLeaderElection,
 			LeaderElectionID:       "b2ed43cb.hypershift.openshift.io",
-			Namespace:              namespace,
 			HealthProbeBindAddress: ":6060",
-			NewClient: func(cache cache.Cache, config *rest.Config, options client.Options, uncachedObjects ...client.Object) (client.Client, error) {
-				// We on operate IngressControllers and Services outside the HCP namespace using the manager client
-				uncachedObjects = append(uncachedObjects, &operatorv1.IngressController{}, &corev1.Service{})
-				return cluster.DefaultNewClient(cache, config, options, uncachedObjects...)
-			},
+			// We manage a service outside the HCP namespace, but we don't want to scope the cache for all objects
+			// to both namespaces so just read from the API.
+			ClientDisableCacheFor: []client.Object{&corev1.Service{}},
+			NewCache: cache.BuilderWithOptions(cache.Options{
```

Comment:
This will still break watching Services in the `openshift-ingress` namespace

Reply:
That is why the `ClientDisableCacheFor` is set for services above, it is a slightly simpler way to do what you did through the custom `NewClient` before and only for services rather than both IngressControllers and Services

Reply:
Ah ok.  I'll test this with private clusters manually tomorrow.